middle_end/flambda2/to_cmm/un_cps_helper.ml

```diff
@@ -294,10 +294,14 @@ let make_block ?(dbg=Debuginfo.none) kind args =
     end;
     make_array ~dbg Naked_floats args
 
-let make_closure_block ?(dbg=Debuginfo.none) l =
+let make_closure_block ?(dbg=Debuginfo.none) l ~has_zero_closures =
   assert (List.compare_length_with l 0 > 0);
-  let tag = Tag.(to_int closure_tag) in
-  make_alloc dbg tag l
+  (* Blocks consisting of only an environment must not have tag Closure_tag *)
+  if has_zero_closures then
```

Comment:
similarly to what is done in the static case, it might be better here to do:
```
let tag = if has_zero_closures then .... in
make_alloc dbg tag l
```
(besides the consistency, it will remove the else branch without parentheses nor `begin..end`, which is nice)